compiler/src/dotty/tools/io/Path.scala

```diff
@@ -208,7 +208,9 @@ class Path private[io] (val jpath: JPath) {
     else {
       Files.walkFileTree(jpath, new SimpleFileVisitor[JPath]() {
         override def visitFile(file: JPath, attrs: BasicFileAttributes) = {
-          Files.delete(file)
+          if (!attrs.isDirectory) {
+            Files.delete(file)
```

Comment:
Looking at the stack trace you posted, this is not the line failing. The line failing is https://github.com/lampepfl/dotty/blob/a0b429bb1220a67fb6331a3615b29ff811bd8147/compiler/src/dotty/tools/io/Path.scala#L216
But the `postVisitDirectory` callback is only called when all files in the directory have been visited and thus deleted. Maybe it is a race condition but then this is not the proper fix